src/newrelic/app_test.go

```diff
@@ -184,6 +184,54 @@ func TestConnectPayloadInternalHostname(t *testing.T) {
 	}
 }
 
+func TestConnectPayloadInternalDocker(t *testing.T) {
+	util := &utilization.Data{}
+	info := &AppInfo{}
+
+	// No docker ID, nil utilization data
+	info.DockerId = ""
+
+	b := info.ConnectPayloadInternal(1, nil)
+
+	result, _ := utilization.GetDockerId(b.Util)
+
+	if nil != b.Util {
+		t.Errorf("expected: %v\nactual: %v", nil, b.Util)
+	}
+
+	// No Docker ID, utilization data
+	info.DockerId = ""
+
+	b = info.ConnectPayloadInternal(1, util)
+
+	result, _ = utilization.GetDockerId(b.Util)
+
+	if result != info.DockerId {
+		t.Errorf("expected: %s\nactual: %v", info.DockerId, result)
+	}
+
+	// Docker ID, nil utilization data
+	info.DockerId = "1056761e1f44969c959364a8e26e9345b37ccb91aef09a8173c90cf1d1d99156"
+
+	b = info.ConnectPayloadInternal(1, nil)
+
+	if nil != b.Util {
+		t.Errorf("expected: %v\nactual: %v", nil, b.Util)
+	}
+
+	// Docker ID, utilization data
```

Comment:
Is this test supposed to have non-nil utilization data?

Reply:
This test passes a non-nil value for the utilization data object. It's testing the expected path for sending the Docker ID in the connect message.

Reply:
Where is `util` set to non-nil? It is not clear in the test case.

Reply:
I see how this code works now - I was overlooking the `nil` arg.